### PR TITLE
Improve build log out but still allow full debug log

### PR DESCRIPTION
**Pull request by jaxesn**
*Issue #, if available:*

*Description of changes:*

This builds on something weve in [build-tooling](https://github.com/aws/eks-distro-build-tooling/blob/main/eks-distro-base/scripts/eks-d-common#L53) to limit the output when building the minimal images, but still allow an override via the `OUTPUT_DEBUG_LOG` env var to see it all if you want.

This PR introduces `build::common::echo_and_run` along with start and end log message for key targets in Common.mk.  echo_and_run was added to a few key places in our common scripts and the `set -x` were removed.  The resulting output, imo, is easier to read and follow and should still provide key data points for when scripts fail. That said, I do expect to see failures in the future that we want to add more logging for. In those case, we can temporary set the `OUTPUT_DEBUG_LOG` env var to see the full debug log.

IMO, set -x is nice because it gives you all the output, but if you are not intimately familiar with and have a strong understanding of what these scripts are doing, its too overwhelming and introduces tons of red herrings.  

Please review the output from some of the presubmit jobs to see what you think and lets discuss!

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Common.mk [hunk 1]
```diff
@@ -221,7 +221,7 @@ trimap = $(and $(strip $2),$(strip $3),$(strip $4),$(call \
 
 _pos = $(if $(filter $1,$2),$(call _pos,$1,\
        $(call list-rem,$2),x $3),$3)
-pos = $(words $(call _pos,$1,$2))
+pos = $(words $(call _pos,$1,$2,))
```
Comment (jaxesn) [inline Common.mk:224]: this was just a long standing bug that only shows it self in a couple builds, and even then its just a warning but its confusing so im fixing it.

**Comment by danbudris** [comment]
I really like this, and I like the idea of having the ability to set `-x` for debugging purposes; we can then add more logging as needed based on waht we find with the debugging enabled.

File: Common.mk [hunk 2]
```diff
@@ -425,6 +425,11 @@ SKIP_CHECKSUM_VALIDATION?=false
 IN_DOCKER_TARGETS=all-attributions all-attributions-checksums all-checksums attribution attribution-checksums binaries checksums clean clean-go-cache
 ####################################################
 
+#################### LOGGING #######################
+TARGET_START_LOG?=$(eval _START_TIME:=$(shell date +%s))\\n------------------- $(shell date +"%Y-%m-%dT%H:%M:%S%z") Starting $@ target -------------------
```
Comment (danbudris) [inline Common.mk:429]: the `$@`, that's an automatic variable which resolves to the name of the target this is executed in, is that right?

Comment (danbudris) [inline Common.mk:429]: should we set `TZ=utc` in here?

Comment (danbudris) [inline Common.mk:429]: some of these targets are super quick, add tenths of a second?

```suggestion
TARGET_START_LOG?=$(eval _START_TIME:=$(shell date +%s))\\n------------------- $(shell date +"%Y-%m-%dT%H:%M:%S:%1N%z") Starting $@ target ------------------- 
```

File: Common.mk [hunk 3]
```diff
@@ -425,6 +425,11 @@ SKIP_CHECKSUM_VALIDATION?=false
 IN_DOCKER_TARGETS=all-attributions all-attributions-checksums all-checksums attribution attribution-checksums binaries checksums clean clean-go-cache
 ####################################################
 
+#################### LOGGING #######################
+TARGET_START_LOG?=$(eval _START_TIME:=$(shell date +%s))\\n------------------- $(shell date +"%Y-%m-%dT%H:%M:%S%z") Starting $@ target ------------------- 
+TARGET_END_LOG?="------------------- `date +'%Y-%m-%dT%H:%M:%S%z'` Finished $@ target (duration `echo $$[( $$(date +%s)-$(_START_TIME) )]` seconds) -------------------\\n"
```
Comment (danbudris) [inline Common.mk:430]:
```suggestion
TARGET_END_LOG?="------------------- `date +'%Y-%m-%dT%H:%M:%S:%1N%z'` Finished $@ target (duration `echo $$[( $$(date +%s)-$(_START_TIME) )]` seconds) -------------------\\n"
```

File: Common.mk (hunk 2 above)
Comment (jaxesn) [reply Common.mk:429]: yep

Comment (jaxesn) [reply Common.mk:429]: will do

File: build/lib/buildkit_check.sh [hunk 4]
```diff
@@ -27,5 +27,6 @@ function buildkit_ready() {
   echo "Buildkit daemon is not available"
   exit 1
 }
+yum install bc -y
```
Comment (jaxesn) [inline build/lib/buildkit_check.sh:30]: i will move this to the builder-base build

File: Common.mk [hunk 5]
```diff
@@ -425,6 +425,13 @@ SKIP_CHECKSUM_VALIDATION?=false
 IN_DOCKER_TARGETS=all-attributions all-attributions-checksums all-checksums attribution attribution-checksums binaries checksums clean clean-go-cache
 ####################################################
 
+#################### LOGGING #######################
+DATE_CMD=TZ=utc $(shell if [ "$$(uname -s)" = "Darwin" ] && command -v gdate &> /dev/null; then echo gdate; else echo date; fi)
+DATE_NANO=$(shell if [ "$$(uname -s)" = "Linux" ] || command -v gdate &> /dev/null; then echo %N; fi)
```
Comment (danbudris) [inline Common.mk:430]: is three decimal places enough?

```suggestion
DATE_NANO=$(shell if [ "$$(uname -s)" = "Linux" ] || command -v gdate &> /dev/null; then echo %3N; fi)
```

File: Common.mk (hunk 2 above)
Comment (danbudris) [reply Common.mk]: hahah that sure is something

File: Common.mk (hunk 5 above)
Comment (jaxesn) [reply Common.mk]: done

**Review by danbudris — APPROVED** [review]

**Review by markapruett — APPROVED** [review]

**Pull request merged**